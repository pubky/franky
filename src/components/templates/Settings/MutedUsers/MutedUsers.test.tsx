import React from 'react';
import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { MutedUsers } from './MutedUsers';

describe('MutedUsers', () => {
  it('renders with default props', () => {
    render(<MutedUsers />);
    expect(screen.getByText('Muted users')).toBeInTheDocument();
  });

  it('renders empty state when no muted users', () => {
    render(<MutedUsers />);
    expect(screen.getByText('No muted users yet')).toBeInTheDocument();
  });

  it('applies custom className', () => {
    const { container } = render(<MutedUsers className="custom-muted" />);
    expect(container.firstChild).toBeInTheDocument();
  });
});

describe('MutedUsers - Snapshots', () => {
  it('matches snapshot with no muted users', () => {
    const { container } = render(<MutedUsers />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with custom className', () => {
    const { container } = render(<MutedUsers className="custom-muted" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
