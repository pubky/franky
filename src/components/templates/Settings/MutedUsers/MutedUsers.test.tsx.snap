// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MutedUsers - Snapshots > matches snapshot with custom className 1`] = `
<div
  class="p-8 md:p-12 bg-[#1D1D20] rounded-md flex-col justify-start items-start gap-12 inline-flex custom-muted"
>
  <div
    class="w-full flex-col justify-start items-start gap-3 flex"
  >
    <div
      class="justify-start items-center gap-2 inline-flex"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-volume-x"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 4.702a.705.705 0 0 0-1.203-.498L6.413 7.587A1.4 1.4 0 0 1 5.416 8H3a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h2.416a1.4 1.4 0 0 1 .997.413l3.383 3.384A.705.705 0 0 0 11 19.298z"
        />
        <line
          x1="22"
          x2="16"
          y1="9"
          y2="15"
        />
        <line
          x1="16"
          x2="22"
          y1="9"
          y2="15"
        />
      </svg>
      <h2
        class="text-2xl font-bold leading-8 text-foreground"
      >
        Muted users
      </h2>
    </div>
    <p
      class="text-base font-medium leading-6 text-secondary-foreground"
    >
      Here is an overview of all users you muted. You can choose to unmute users if you want.
    </p>
    <div
      class="w-full flex-col inline-flex gap-3"
    >
      <h2
        class="text-2xl font-normal leading-8 opacity-20 text-center flex self-center mt-5"
      >
        No muted users yet
      </h2>
    </div>
  </div>
</div>
`;

exports[`MutedUsers - Snapshots > matches snapshot with no muted users 1`] = `
<div
  class="p-8 md:p-12 bg-[#1D1D20] rounded-md flex-col justify-start items-start gap-12 inline-flex"
>
  <div
    class="w-full flex-col justify-start items-start gap-3 flex"
  >
    <div
      class="justify-start items-center gap-2 inline-flex"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-volume-x"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 4.702a.705.705 0 0 0-1.203-.498L6.413 7.587A1.4 1.4 0 0 1 5.416 8H3a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h2.416a1.4 1.4 0 0 1 .997.413l3.383 3.384A.705.705 0 0 0 11 19.298z"
        />
        <line
          x1="22"
          x2="16"
          y1="9"
          y2="15"
        />
        <line
          x1="16"
          x2="22"
          y1="9"
          y2="15"
        />
      </svg>
      <h2
        class="text-2xl font-bold leading-8 text-foreground"
      >
        Muted users
      </h2>
    </div>
    <p
      class="text-base font-medium leading-6 text-secondary-foreground"
    >
      Here is an overview of all users you muted. You can choose to unmute users if you want.
    </p>
    <div
      class="w-full flex-col inline-flex gap-3"
    >
      <h2
        class="text-2xl font-normal leading-8 opacity-20 text-center flex self-center mt-5"
      >
        No muted users yet
      </h2>
    </div>
  </div>
</div>
`;
